### ISSUE TITLE
mini-batch g.d. and Kmeans

### ISSUE THREAD

**Issue by majorli** [majorli/dl#4]
- Mini-batch gradient descent
- K-means

**Comment by majorli** [comment]
mini-batch g.d. and Adam algo. are completed.
K-means in progress

**Comment by majorli** [comment]
Changing classes structure, implement Batch Normalization.
In progress...
==== 2017/12/29 11:20 =====
To test:
NNModel.feed_data()

**Comment by majorli** [comment]
completed, 20180105
